### PR TITLE
Fix issue #119

### PR DESCRIPTION
**Pull request by tuliom**
This patch set implements the requests from PR #120 and fixes issue #119:

- r2 is saved and restored manually.
- Added a ChangeLog to the file.
- Added r30 to the list of registers to save.

### CODE REVIEW

**Comment by tuliom** [comment]
Notice that older GCC versions (<= 6) have a bug that w fixed with [rev. 235876](https://gcc.gnu.org/git/?p=gcc.git;a=commitdiff;h=9213244550335bcb2b8590a0d7d58ac74c932361).
You may need to fix that bug in order to clobber r30.

**Comment by tuliom** [comment]
PR #119 can be closed with this PR.

**Comment by snaury** [comment]
Thank you!

File: platform/switch_ppc64_linux.h [hunk 1]
```diff
@@ -2,6 +2,12 @@
  * this is the internal transfer function.
  *
  * HISTORY
+ * 21-Mar-18  Tulio Magno Quites Machado Filho  <tuliom@linux.vnet.ibm.com>
+ *      Added r30 to the list of saved registers in order to fully comply with
+ *      both ppc64 ELFv1 ABI and the ppc64le ELVv2 ABI, that classify this
```
Comment (arcivanov) [inline platform/switch_ppc64_linux.h]: ELFv2

Comment (snaury) [reply platform/switch_ppc64_linux.h]: Thanks! Fixed.

Comment (tuliom) [reply platform/switch_ppc64_linux.h]: Ooops.  Thanks for fixing and merging it!

**Pull request merged**